ledger-core/virtual/integration/deduplication/constructor_test.go

```diff
@@ -139,3 +141,206 @@ func TestConstructor_SamePulse(t *testing.T) {
 		})
 	}
 }
+
+func TestDeduplication_DifferentPulses(t *testing.T) {
+	isolation := contract.ConstructorIsolation()
+
+	tests := []struct {
+		Name       string
+		TestRailID string
+		Skipped    string
+
+		VState           payload.VStateReport
+		VStateSendBefore bool
+		VFindCall        *payload.VFindCallResponse
+		VDelegatedCall   *payload.VDelegatedCallRequest
+	}{
+		{
+			Name:       "status is unknown",
+			TestRailID: "C5006",
+			Skipped:    "https://insolar.atlassian.net/browse/PLAT-386",
+			VState: payload.VStateReport{
+				Status:              payload.Empty,
+				OrderedPendingCount: 1,
+			},
+			VStateSendBefore: false,
+			VFindCall: &payload.VFindCallResponse{
+				Status:     payload.UnknownCall,
+				CallResult: nil,
+			},
+		},
+		{
+			Name:       "status is found and result is empty",
+			TestRailID: "C5007",
+			VState: payload.VStateReport{
+				Status:              payload.Empty,
+				OrderedPendingCount: 1,
+			},
+			VStateSendBefore: false,
+			VFindCall: &payload.VFindCallResponse{
+				Status:     payload.FoundCall,
+				CallResult: nil,
+			},
+		},
+		{
+			Name:       "status is found and result is non-empty",
+			TestRailID: "C5008",
+			VState: payload.VStateReport{
+				Status:              payload.Empty,
+				OrderedPendingCount: 1,
+			},
+			VStateSendBefore: false,
+			VFindCall: &payload.VFindCallResponse{
+				Status:     payload.FoundCall,
+				CallResult: &payload.VCallResult{ReturnArguments: []byte("123")},
+			},
+		},
+		{
+			Name:       "status is empty and pending is known",
+			TestRailID: "C5012",
+			VState: payload.VStateReport{
+				Status:              payload.Empty,
+				OrderedPendingCount: 1,
+				ProvidedContent:     &payload.VStateReport_ProvidedContentBody{},
+			},
+			VStateSendBefore: true,
+			VDelegatedCall: &payload.VDelegatedCallRequest{
+				Callee:       payload.Reference{},
+				CallOutgoing: payload.Reference{},
+				CallFlags:    payload.BuildCallFlags(isolation.Interference, isolation.State),
+			},
+		},
+	}
+	for _, test := range tests {
+		t.Run(test.Name, func(t *testing.T) {
+			t.Log(test.TestRailID)
+			if test.Skipped != "" {
+				t.Skip(test.Skipped)
+			}
+
+			mc := minimock.NewController(t)
+
+			server, ctx := utils.NewUninitializedServer(nil, t)
+			defer server.Stop()
+
+			executeDone := server.Journal.WaitStopOf(&execute.SMExecute{}, 1)
+
+			runnerMock := logicless.NewServiceMock(ctx, mc, nil)
+			server.ReplaceRunner(runnerMock)
+			server.Init(ctx)
+
+			var (
+				class         = gen.UniqueGlobalRef()
+				previousPulse = server.GetPulse().PulseNumber
+				outgoingLocal = gen.UniqueLocalRefWithPulse(previousPulse)
+				outgoing      = reference.NewRecordOf(server.GlobalCaller(), outgoingLocal)
+				object        = outgoing
+			)
+
+			test.VState.Object = object
+			test.VState.OrderedPendingEarliestPulse = previousPulse
+			if test.VState.ProvidedContent != nil {
+				test.VState.ProvidedContent.OrderedQueue = []reference.Global{outgoing}
```

Comment:
what is this? Ordered queue is not implemented in R0.

Reply:
fixed